### DISCUSSION TITLE
The Node.js native addon module (tfjs_binding.node) can not be found

### DISCUSSION THREAD

**Discussion by YashKumarVerma** [camicroscope/Caracal#81] [Q&A] [answered]
So I faced this problem that said that the nation addon was not found. 

The error logs ask to rebuild the native package, [check image]
![https://i.imgur.com/1EKeN0T.png](https://i.imgur.com/1EKeN0T.png) 

As expected, running it rendered one more error.
![https://i.imgur.com/zoW87YC.png](https://i.imgur.com/zoW87YC.png)

So after some research, I've found that this particular error throws only if the node is running on v15. **Dropping the version to v14 has fixed the problem**. I strongly recommend using nvm to handle cases like these.

Finally built and running 
![https://i.imgur.com/kUFFOC2.png](https://i.imgur.com/kUFFOC2.png)

## TLDR
- drop to node v14 to build the TF package.
- when doing a fresh install, make sure you run `npm rebuild @tensorflow/tfjs-node build-addon-from-source`

**Comment by birm** [comment, marked as answer]
I forgot we added this feature. I wonder if there's any tenuous link to https://github.com/camicroscope/Caracal/issues/44

**Reply by siddhanth339** [reply]
Thanks, it helped!!!
Is there any reason to drop to node v14 and build the TF package?
